python/taichi/lang/matrix.py

```diff
@@ -772,19 +763,24 @@ def empty(n, m):
         mat = ti.Matrix([[None] * m for _ in range(n)])
         return mat
 
+    @staticmethod
+    def new(n, m):
+        if impl.inside_kernel():
+            return Matrix(n, m)
+        else:
+            return Matrix.empty(n, m)
```

Comment:
🤔 The use cases of `new` looked more like this would be a `classmethod` to me, but I'm not 100% sure. e.g.

```suggestion
    @classmethod
    def new(cls, n, m):
        if impl.inside_kernel():
            return cls(n, m)
        else:
            return cls.empty(n, m)
```

Reply:
Thank for teaching me Python knowledge! I never know this `@classmethod` decorator before, it looks so useful, thanks! @yuanming-hu Will we make other initializers like `Matrix.cols/rows/var` a class method too? Or `TaichiClass.var` should be?